store/tikv/lock_resolver.go

```diff
@@ -327,8 +334,15 @@ func (lr *LockResolver) ResolveLocks(bo *Backoffer, callerStartTS uint64, locks
 			// Update the txn expire time.
 			msBeforeLockExpired := lr.store.GetOracle().UntilExpired(l.TxnID, status.ttl)
 			msBeforeTxnExpired.update(msBeforeLockExpired)
-			// In the write conflict scenes, callerStartTS is set to 0 to avoid unnecessary push minCommitTS operation.
-			if callerStartTS > 0 {
+			if forWrite {
+				// Write conflict detected!
+				// If it's a optimistic conflict and current txn is later than the lock owner,
+				// abort current transaction.
+				// This could avoids the deadlock scene of two large transaction.
+				if l.LockType == kvrpcpb.Op_Put && l.TxnID > callerStartTS {
```

Comment:
I think it should abort with any `LockType`.

Reply:
1. The pessimistic lock might rollback, maybe we retry and the pessimistic lock has gone.
2. The pessimistic lock could turn into an optimistic lock in the prewrite phase, so the conflict could be detected anyway